waku/v2/node/peer_manager/waku_peer_store.nim

```diff
@@ -55,6 +56,10 @@ proc canBeConnected*(peerStore: PeerStore,
   if failedAttempts == 0:
     return true
 
+  # if there are too many failed attemps, do not reconnect
+  if failedAttempts >= maxFailedAttempts:
+    return false
```

Comment:
we should probably add a check earlier to ensure that the maxFailedAttempts is < 14